server/data/restClient.ts

```diff
@@ -83,16 +84,17 @@ export default class RestClient {
       })
   }
 
-  async post({
+  async update({
```

Comment:
@nathanrussell-moj-digital - I wrote a put function that looked exactly the same as the post function so I merged them into one. I had no idea what to call this new function so I went with update. I don't like it. Do you have any better ideas?

Reply:
Given that this class is `restClient` I would say there is an assumed level of knowledge as to what someone is trying to do, so using 'rest' (HTTP) terms doesn't feel unreasonable to me. How about taking inspiration('ish) from Spring's `RestTemplate` as conceptually this class is not a million miles away from it?

How about 3 public methods - `get`, `post` and `put`. Given that the post and the put are so similar then I agree, you want to refactor in some way. And then a private method `exchange` (which is basically your proposed `update` method). Your 2 public `post` and `put` methods could then just delegate to the private `exchange` method, passing in the required HTTP verb.

Would that work?